pkg/collector/corechecks/oracle-dbm/oracle.go

```diff
@@ -200,9 +217,38 @@ func (c *Check) Connect() (*sqlx.DB, error) {
 		}
 	}
 
+	if c.config.AgentSQLTrace.Enabled {
+		db.SetMaxOpenConns(1)
+		_, err := db.Exec("ALTER SESSION SET tracefile_identifier='DDAGENT'")
+		if err != nil {
+			log.Warnf("failed to set tracefile_identifier: %v", err)
+		}
+
+		/* We are concatenating values instead of passing parameters, because there seems to be a problem
+		 * in go-ora with passing bool parameters to PL/SQL. As a mitigation, we are asserting that the
+		 * parameters are bool
+		 */
+		binds := returnFalseIfNotBool(c.config.AgentSQLTrace.Binds)
+		waits := returnFalseIfNotBool(c.config.AgentSQLTrace.Waits)
+		setEventsStatement := fmt.Sprintf("BEGIN dbms_monitor.session_trace_enable (binds => %t, waits => %t); END;", binds, waits)
```

Comment:
By the way, `go vet` checks if the parameters to `fmt.Sprintf()` match the types of the formatter. I'm pretty sure that `go vet` is part of our CI checks but if not, the result for the wrong type passed for binds and waits would be this string:
```
"BEGIN dbms_monitor.session_trace_enable (binds => %!t(string=SOMEONE TRYING TO INJECT HERE), waits => %!t(string=SOMEONE TRYING TO INJECT HERE)); END;
```
I think the way the `%!t` gets inserted would definitely make it harder to try and inject something but I just wanted to bring this up so we're all clear what the level of risk is. 

[Playground example](https://go.dev/play/p/88-Gw3qdP02). Note that I had to obfuscate the format passed to `fmt.Sprintf` by putting it in a variable because the playground runs `go vet` by default and it wouldn't let me run the code otherwise.

Reply:
Yes, your solution is much more elegant. Implemented now, thanks.

Reply:
Thanks. And the last line of defence is the parameter checking by the Oracle `dbms_monitor.session_trace_enable` procedure:

```BEGIN dbms_monitor.session_trace_enable (binds => 'select sysdate from dual'); end;
      *
ERROR at line 1:
ORA-06550: line 1, column 7:
PLS-00306: wrong number or types of arguments in call to 'SESSION_TRACE_ENABLE'
ORA-06550: line 1, column 7:
PL/SQL: Statement ignored```